library/src/contraction/contraction_solution_impl.hpp

```diff
@@ -232,10 +389,723 @@ namespace hiptensor
+                      std::vector<std::size_t> const& ds_ms_ns_lengths,
+                      std::vector<std::size_t> const& ds_ms_ns_strides,
+                      std::vector<std::size_t> const& e_ms_ns_lengths,
+                      std::vector<std::size_t> const& e_ms_ns_strides,
+                      void*                           workspacePtr) override
+        {
+            using Traits = MetaTraits<DeviceOp>;
+
+            // Clear out the previous arguments
+            Base::resetArgs();
+
+            // Promote to derived class for necessary functions such as
+            // MakeArgumentPointer and MakeInvokerPointer.
+            auto* deviceOp = dynamic_cast<DeviceOp*>(mDeviceOp.get());
+
+            // Note: CK ALWAYS uses float for alpha / beta in contraction multipleD
+            auto alphaF = 0.0f;
+            auto betaF  = 0.0f;
+
+            if(alpha != nullptr)
+            {
+                alphaF = hiptensor::readVal<float>(alpha, convertToComputeType(HipDataType_v<typename Traits::EDataT>));
+            }
+            if(beta != nullptr)
+            {
+                betaF = hiptensor::readVal<float>(beta, convertToComputeType(HipDataType_v<typename Traits::EDataT>));
+            }
+
+            // CK has its own format for indices...
+            auto toCKVec = [](std::vector<std::size_t> const& v) {
+                return std::vector<ck::index_t>(v.begin(), v.end());
+            };
+
+            //Allocate Real and Imaginary inputs
+            auto elementsA  = elementSpaceFromLengthsAndStrides(a_ms_ks_lengths, a_ms_ks_strides);
+            auto elementsB  = elementSpaceFromLengthsAndStrides(b_ns_ks_lengths, b_ns_ks_strides);
+            auto elementsD  = elementSpaceFromLengthsAndStrides(ds_ms_ns_lengths, ds_ms_ns_strides);
+            elementsE       = elementSpaceFromLengthsAndStrides(e_ms_ns_lengths, e_ms_ns_strides);
+
+            A_d_real.reset(nullptr);
+            A_d_real = std::move(allocDevice<elementType>(elementsA));
+            A_d_imag.reset(nullptr);
+            A_d_imag = std::move(allocDevice<elementType>(elementsA));
+            B_d_real.reset(nullptr);
+            B_d_real = std::move(allocDevice<elementType>(elementsB));
+            B_d_imag.reset(nullptr);
+            B_d_imag = std::move(allocDevice<elementType>(elementsB));
+            D_d_real.reset(nullptr);
+            D_d_real = std::move(allocDevice<elementType>(elementsD));
+            D_d_imag.reset(nullptr);
+            D_d_imag = std::move(allocDevice<elementType>(elementsD));
+            E_d_real.reset(nullptr);
+            E_d_real = std::move(allocDevice<elementType>(elementsE));
+            E_d_imag.reset(nullptr);
+            E_d_imag = std::move(allocDevice<elementType>(elementsE));
+
+            auto blockDim = dim3(1024, 1, 1);
+            auto gridDim  = dim3(ceilDiv(elementsA, blockDim.x), 1, 1);
+            hipLaunchKernelGGL(
```

Comment:
We usually use hipLaunchKernelGGL if we need timing params through events. Can get away with the triple chevron syntax instead <<<>>>